resources/assets/js/device.js

```diff
@@ -0,0 +1,26 @@
+$(document).ready(function () {
```

Comment:
Also, please rename this file to something more descriptive like `UniqueDeviceNameValidator.js`

Reply:
It can be renamed and it only contains one validator now. But I don't think this naming is a good idea. Usually we don't separate each validator in a js file. You probably will have another validator and you may put it in the same js file in the future.

Reply:
What about `DeviceValidators.js` then?  I just feel like the name of the file needs a little more detail.